server/pulp/server/managers/content/orphan.py

```diff
@@ -256,13 +266,32 @@ def delete_orphan_content_units_by_type(type_id):
             for non_orphan_id in non_orphan:
                 unit_dict.pop(non_orphan_id)
 
-            # Remove the unit and any references on disk
+            # Remove the unit, lazy catalog entries, and any content in storage.
             for unit_to_delete in unit_dict.itervalues():
+                OrphanManager.delete_catalog_entry(unit_to_delete.id, type_id)
```

Comment:
What if you replace this line with:

```
model.LazyCatalogEntry.objects(unit_id=unit_id, unit_type_id=unit_type_id).delete()
```

and get rid of the function below? I don't think the extra function makes things easier.